### PR TITLE
[tune] LightGBM integration callbacks

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

This PR adds LightGBM callbacks for Tune integration, working in a similar way to XGBoost ones, and updates the existing example to make use of them.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by krfricke** [comment]
Can we turn these into (callable) classes? That way it's easier to extend/mock them and it would follow the other integrations.

Also, should we add a test for this?

**Comment by Yard1** [comment]
@krfricke There weren't any tests I could find for the xgboost integration, so I imagine the example itself suffices as a test. Though it should probably test both callbacks.

LighGBM callbacks are usually done like this but you are right, a callable class should work just as well while being easier to maintain. I'll see if they can be converted to one.

**Comment by krfricke** [comment]
Ok, examples will do then.

Yes, a class would be nice - we do this for some other function-based callback integrations as well

File: python/ray/tune/integration/lightgbm.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            config,
+            train_set,
+            valid_sets=[test_set],
+            valid_names=["eval"],
+            verbose_eval=False,
+            callbacks=[TuneReportCheckpointCallback(
+                {"loss": "eval-binary_logloss"}, "lightgbm.mdl)])
+
+    """
+    order = 21
```
Comment (richardliaw) [inline python/ray/tune/integration/lightgbm.py:162]: what does `order` do?

Comment (Yard1) [reply python/ray/tune/integration/lightgbm.py:162]: It's a LightGBM property. Callbacks with lower order will be ran first. It's a minor thing in this case but it's been set for completeness, as all built-in LGBM callbacks have it.

**Review by krfricke — APPROVED** [review]
Looks great, thanks!

**Comment by Yard1** [comment]
@richardliaw Can you merge this? Thanks!

**Pull request merged**